File: src/test/java/com/d20charactersheet/framework/dac/dao/BaseFeatDaoTest.java
package com.d20charactersheet.framework.dac.dao;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.List;

import org.junit.Test;

import com.d20charactersheet.framework.boc.model.Character;
import com.d20charactersheet.framework.boc.model.CharacterFeat;
import com.d20charactersheet.framework.boc.model.Feat;

public abstract class BaseFeatDaoTest {

  protected CharacterDao characterDao;
  protected FeatDao featDao;

  @Test
  public void testGetAllFeats() {

    // Act
    final List<Feat> allFeats = featDao.getAllFeats();

    // Assert
    assertThat(allFeats).hasSize(109);
  }

  @Test
  public void testGetFeatsOfCharacter() {
    // Arrange
    final Character belvador = new Character();
    belvador.setId(0);

    // Act
    final List<CharacterFeat> characterFeats = characterDao.getCharacterFeats(belvador, featDao.getAllFeats());

    // Assert
    assertThat(characterFeats).hasSize(3);
  }

}

File: src/test/java/com/d20charactersheet/framework/dac/dao/BaseFeatDaoTest.kt
package com.d20charactersheet.framework.dac.dao

import com.d20charactersheet.framework.boc.model.Character
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

abstract class BaseFeatDaoTest {

    protected lateinit var characterDao: CharacterDao
    protected lateinit var featDao: FeatDao

    @Test
    fun testGetAllFeats() {

        // Act
        val allFeats = featDao.getAllFeats()

        // Assert
        assertThat(allFeats).hasSize(109)
    }

    @Test
    fun testGetFeatsOfCharacter() {
        // Arrange
        val belvador = Character()
        belvador.id = 0

        // Act
        val characterFeats = characterDao.getCharacterFeats(belvador, featDao.getAllFeats())

        // Assert
        assertThat(characterFeats).hasSize(3)
    }
}